component/common/net/config.go

```diff
@@ -3,30 +3,62 @@
 package net
 
 import (
-	"flag"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/middleware"
 	"time"
-
-	weaveworks "github.com/weaveworks/common/server"
 )
 
 const (
 	DefaultHTTPPort = 8080
-	DefaultGRPCPort = 8081
 )
 
-// ServerConfig is a River configuration that allows one to configure a weaveworks.Server. It
-// exposes a subset of the available configurations.
+// ServerConfig for a Server
 type ServerConfig struct {
```

Comment:
This struct is a merge of our ServerConfig and the weaveworks' server Config.
But we would be in control of how this works and what is available in River.